CMakeLists.txt

```diff
@@ -52,6 +52,33 @@ endif ()
 
 include(FetchContent)
 
+option(SALTATLAS_USE_METALL "Use Metall" OFF)
+
+#
+# Boost
+#
+find_package(Boost 1.64 QUIET)
+if (NOT Boost_FOUND)
+    FetchContent_Declare(Boost
+            URL https://boostorg.jfrog.io/artifactory/main/release/1.79.0/source/boost_1_79_0.tar.bz2)
+    FetchContent_GetProperties(Boost)
+    if (Boost_POPULATED)
+        message(STATUS ${PROJECT_NAME}
+                " found already populated Boost dependency: "
+                ${boost_SOURCE_DIR})
+    else ()
+        if (SALTATLAS_USE_METALL) # Only Metall requires Boost
```

Comment:
Please note that the current implementation downloads Boost only when Metall is requested since that's the only library that requires Boost.